src/test/java/com/musala/atmosphere/websocket/util/UiElementSelectorDeserializationTest.java

```diff
@@ -0,0 +1,124 @@
+package com.musala.atmosphere.websocket.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.musala.atmosphere.commons.RoutingAction;
+import com.musala.atmosphere.commons.geometry.Bounds;
+import com.musala.atmosphere.commons.geometry.Point;
+import com.musala.atmosphere.commons.ui.selector.CssAttribute;
+import com.musala.atmosphere.commons.ui.selector.UiElementSelectionOption;
+import com.musala.atmosphere.commons.ui.selector.UiElementSelector;
+import com.musala.atmosphere.commons.websocket.message.RequestMessage;
+
+/**
+ * Tests the deserialization of the {@link UiElementSelector} data.
+ *
+ * @author dimcho.nedev
+ *
+ */
+public class UiElementSelectorDeserializationTest extends CustomDataDeserializationTest {
+
+    @Test
+    public void uiElementSelectorDeserializatinTest() {
+        String jsonMessage = "{\"arguments\":[{\"mKey\":\"com.musala.atmosphere.commons.ui.selector.UiElementSelector\","
+                + "\"mValue\":{\"attributeProjectionMap\":{\"CLASS_NAME\":{\"mKey\":\"android.widget.NumberPicker\","
+                + "\"mValue\":\"EQUALS\"},\"INDEX\":{\"mKey\":1,\"mValue\":\"EQUALS\"}}}},{\"mKey\":\"java.lang.Boolean\",\"mValue\":true}],"
+                + "\"messageAction\":\"ROUTING_ACTION\",\"routingAction\":\"GET_UI_ELEMENTS\","
+                + "\"deviceId\":\"59A57D012B068B4B0D227342EEA9262E_01d04ee91317a4d6\",\"sessionId\":\"726363d4-23b0-4125-b5cf-f411c3d89139\"}";
+
+        RequestMessage requestMessage = jsonUtil.deserializeRequest(jsonMessage);
+
+        UiElementSelector uiElementSelector = (UiElementSelector) (requestMessage.getArguments()[0]);
+
+        Assert.assertEquals(uiElementSelector.getClassName(), "android.widget.NumberPicker");
```

Comment:
No. This is the name of the CLASS_NAME attribute, not the name of the uiElementSelector class.